core/cap-0050.md

```diff
@@ -0,0 +1,183 @@
+```
+CAP: 0050
+Title: Smart Contract Signatures
+Working Group:
+    Owner: Jonathan Jove <@jonjove>
+    Authors:
+    Consulted:
+Status: Draft
+Created: 2022-05-10
+Discussion: TBD
+Protocol version: TBD
+```
+
+## Simple Summary
+
+This proposal explains how to authorize a smart contract interaction.
+
+## Motivation
+
+Actions in smart contracts must be authorized by an external entity. The
+authorization method can vary by smart contract.
+
+Basic authorization provides the authority to interact with a smart contract in
+the name of an account. For example, in CAP-48 and CAP-49 this level of
+authorization is sufficient to transfer funds.
+
+A more sophisticated aspect of authorization is the ability for one account to
+grant authority to another account or contract. In some cases, it is sufficient
+for the grantor to take an on-chain action. For example, in CAP-48 and CAP-49
+`approve`, which depends on basic authorization, can be used to grant control
+over the grantor's funds. In other cases, it is preferable for the grantee to
+take the on-chain action. This is demonstrated by `permit` in EIP-2612. When the
+grantee takes the on-chain action, the grantor must have presigned a message
+which conveys the relevant authority.
+
+This proposal is concerned with basic authorization and presigned messages.
+While developers are free to adopt a different standard for presigned messages,
+presigned messages will likely make an appearance in asset interoperability so
+it is worth choosing something appropriate now.
+
+### Requirements
+
+#### Basic Authorization is Simple
+
+Basic authorization, by its very nature, is intended to be basic. With smart
+contracts, it is always possible to build a smart wallet with more sophisticated
+capabilities. This approach has been employed successfully on Ethereum, where
+most accounts use only basic authorization but some high value accounts are
+secured by a smart wallet.
+
+#### Domain Separation
+
+To ensure security, it is essential to understand exactly what you are signing
+and to interpret that data in every context where the signature confers
+authority. As a cryptographic key can be used anywhere, there is no way to
+design a protocol which ensures that no unintented authority is ever granted.
+This problem can often be simplified by only using a cryptographic key within a
+single context. In this case, you simply interpret the data in that single
+context.
+
+In the context of presigned messages, it is generally impossible to ensure that
+the cryptographic key is only used in a single context. The reason is that the
+same key is used for basic authorization and presigned messages. In this case,
+the signature for a carefully crafted presigned message could in fact authorize
+a transaction, or the signature for a carefully crafted transaction could be a
+valid presigned message in some relevant domain.
```

Comment:
This would only be possible if the presigned message literally was a transaction hash, right?